shards/watcher.go

```diff
@@ -55,9 +55,11 @@ func NewDirectoryWatcher(dir string, loader shardLoader) (io.Closer, error) {
 		loader:     loader,
 		quit:       quitter,
 	}
-	if err := sw.scan(); err != nil {
-		return nil, err
-	}
+	go func() {
+		if err := sw.scan(); err != nil {
+			log.Fatalf("failed to scan index directory: %v", err)
+		}
+	}()
```

Comment:
it isn't safe to do scan and watch concurrently since watch calls scan. I suggest:

- move sw.scan inside the goroutine that watch calls.
- The only error that scan can return is from readdir. This doesn't seem that important to report, so we don't need to worry about returning its error.

Reply:
Done.